crates/ra_ide_db/src/symbol_index.rs

```diff
@@ -173,6 +176,30 @@ pub fn world_symbols(db: &RootDatabase, query: Query) -> Vec<FileSymbol> {
     query.search(&buf)
 }
 
+pub fn crate_symbols(db: &RootDatabase, krate: CrateId, query: Query) -> Vec<FileSymbol> {
+    let def_map = db.crate_def_map(krate);
+    let mut files = Vec::new();
+    let mut modules = vec![def_map.root];
+    while let Some(module) = modules.pop() {
+        let data = &def_map[module];
+        files.extend(data.origin.file_id());
+        modules.extend(data.children.values());
+    }
```

Comment:
Hm, if we are looking at `def_map` anyway, maybe we should just dump the symbols from it directly (as opposed to collecting the files, and *then* getting symbols from files' syntax)?

Reply:
@jonas-schievink are we leaving this for the future work?

Reply:
I think so, yes. I can put in a FIXME or open an issue.

Reply:
r=me with an issue linked!